Doc/extending/embedding.rst

```diff
@@ -275,6 +275,91 @@ will depend on the details of the C++ system used; in general you will need to
+frozen and use ``PyRun_SimpleString`` that contains their main module(s) but
+also depend on an frozen module (like the ``__hello__`` module). However if we
+wanted to add our own module to the frozen list we would normally do this::
+
+    #include <importlib.h>
+    #include <importlib_external.h>
+    #include "mymodule.h"
+
+    static const struct _frozen _PyImport_FrozenModules[] = {
+        /* importlib */
+        {"_frozen_importlib", _Py_M__importlib, (int)sizeof(_Py_M__importlib)},
+        {"_frozen_importlib_external", _Py_M__importlib_external,
+            (int)sizeof(_Py_M__importlib_external)},
+        /* mymodule */
+       {"mymodule", M_mymodule, (int)sizeof(M_mymodule)},
+       {0, 0, 0} /* sentinel */
+    };
+
+    const struct _frozen * PyImport_FrozenModules = _PyImport_FrozenModules;
+
+.. note::
+   The reason why this example includes ``importlib.h`` and
+   ``importlib_external.h`` is because of the fact that your embedded
+   interpreter might need them to run properly (load the python standard
+   library from an zip file or the sources like normally). Also in order
+   for your compiler to work you must point it to the Python subdirectory
+   in an active clone of the cpython repository or source tarball.
+   However it is not required to manually build the python core if you are
+   using an clone or source tarball of an version installed on your system.
+
+As you can see the above code will compile (with an warning on Windows
+which will export ``PyImport_FrozenModules`` on the embedded python program).
+This is not what we want. And if we were to run it with ``myprogram`` We will
+get this traceback on running the string based main module(s).
+
+.. code-block:: py
+
+    Traceback (most recent call last):
+      File "<string>", line 1, in <module>
+    ModuleNotFoundError: No module named 'mymodule'
+
+There is got to be a better way to have what we want but to also make python
+aware of our ``mymodule`` being an frozen module. This module could be
+anything from an import hook or anything else where you want to freeze
+it similar to how ``importlib`` is frozen to support how your code base
+currently is.
+
+The fix to this is to change the C code above to::
+
+    #include <importlib.h>
+    #include <importlib_external.h>
+    #include "mymodule.h"
+
+    static const struct _frozen _PyImport_FrozenModules[] = {
+        /* importlib */
+        {"_frozen_importlib", _Py_M__importlib, (int)sizeof(_Py_M__importlib)},
+        {"_frozen_importlib_external", _Py_M__importlib_external,
+            (int)sizeof(_Py_M__importlib_external)},
+        /* mymodule */
+       {"mymodule", M_mymodule, (int)sizeof(M_mymodule)},
+       {0, 0, 0} /* sentinel */
+    };
+
+And then in the main() C or C++ function in your embedded interpreter add this line::
+
+    PyImport_FrozenModules = _PyImport_FrozenModules;
+
+Now your Embedded python *should* be able to load your frozen modules perfectly fine.
```

Comment:
No need to emphasize the 'should', unless there's a specific reason I'm missing.

```suggestion
Now your embedded Python should be able to load your frozen modules perfectly fine.
```